File: openapi/java/src/main/java/org/openapitools/client/model/V3DynamoDbTimetable.java
/*
 * PTV Timetable API - Version 3
 * The PTV Timetable API provides direct access to Public Transport Victoria's public transport timetable data.    The API returns scheduled timetable, route and stop data for all metropolitan and regional train, tram and bus services in Victoria, including Night Network(Night Train and Night Tram data are included in metropolitan train and tram services data, respectively, whereas Night Bus is a separate route type).    The API also returns real-time data for metropolitan train, tram and bus services (where this data is made available to PTV), as well as disruption information, stop facility information, and access to myki ticket outlet data.    This Swagger is for Version 3 of the PTV Timetable API. By using this documentation you agree to comply with the licence and terms of service.    Train timetable data is updated daily, while the remaining data is updated weekly, taking into account any planned timetable changes (for example, due to holidays or planned disruptions). The PTV timetable API is the same API used by PTV for its apps. To access the most up to date data PTV has (including real-time data) you must use the API dynamically.    You can access the PTV Timetable API through a HTTP or HTTPS interface, as follows:        base URL / version number / API name / query string  The base URL is either:    *  http://timetableapi.ptv.vic.gov.au  or    *  https://timetableapi.ptv.vic.gov.au    The Swagger JSON file is available at http://timetableapi.ptv.vic.gov.au/swagger/docs/v3    Frequently asked questions are available on the PTV website at http://ptv.vic.gov.au/apifaq    Links to the following information are also provided on the PTV website at http://ptv.vic.gov.au/ptv-timetable-api/  * How to register for an API key and calculate a signature  * PTV Timetable API V2 to V3 Migration Guide  * PTV Timetable API Data Quality Statement    All information about how to use the API is in this documentation. PTV cannot provide technical support for the API.  
 *
 * The version of the OpenAPI document: v3
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */


package org.openapitools.client.model;

import java.util.Objects;
import com.google.gson.TypeAdapter;
import com.google.gson.annotations.JsonAdapter;
import com.google.gson.annotations.SerializedName;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonWriter;
import java.io.IOException;
import java.util.Arrays;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import com.google.gson.JsonArray;
import com.google.gson.JsonDeserializationContext;
import com.google.gson.JsonDeserializer;
import com.google.gson.JsonElement;
import com.google.gson.JsonObject;
import com.google.gson.JsonParseException;
import com.google.gson.TypeAdapterFactory;
import com.google.gson.reflect.TypeToken;
import com.google.gson.TypeAdapter;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonWriter;
import java.io.IOException;

import java.lang.reflect.Type;
import java.util.HashMap;
import java.util.HashSet;
import java.util.List;
import java.util.Map;
import java.util.Set;

import org.openapitools.client.JSON;

/**
 * V3DynamoDbTimetable
 */
@javax.annotation.Generated(value = "org.openapitools.codegen.languages.JavaClientCodegen", date = "2024-04-22T17:23:48.737568700+10:00[Australia/Sydney]", comments = "Generator version: 7.5.0")
public class V3DynamoDbTimetable {
  public static final String SERIALIZED_NAME_TABLE_NAME = "table_name";
  @SerializedName(SERIALIZED_NAME_TABLE_NAME)
  private String tableName;

  public static final String SERIALIZED_NAME_PARSER_VERSION = "parser_version";
  @SerializedName(SERIALIZED_NAME_PARSER_VERSION)
  private Long parserVersion;

  public static final String SERIALIZED_NAME_PARSER_MAPPING_VERSION = "parser_mapping_version";
  @SerializedName(SERIALIZED_NAME_PARSER_MAPPING_VERSION)
  private String parserMappingVersion;

  public static final String SERIALIZED_NAME_PT_VERSION = "pt_version";
  @SerializedName(SERIALIZED_NAME_PT_VERSION)
  private Long ptVersion;

  public static final String SERIALIZED_NAME_PT_MAPPING_VERSION = "pt_mapping_version";
  @SerializedName(SERIALIZED_NAME_PT_MAPPING_VERSION)
  private String ptMappingVersion;

  /**
   * A.k.a. Transport Mode (e.g. Train, Tram, Bus, V/Line, Nightrider)
   */
  @JsonAdapter(TransportTypeEnum.Adapter.class)
  public enum TransportTypeEnum {
    NUMBER_0(0),
    
    NUMBER_1(1),
    
    NUMBER_2(2),
    
    NUMBER_3(3),
    
    NUMBER_4(4);

    private Integer value;

    TransportTypeEnum(Integer value) {
      this.value = value;
    }

    public Integer getValue() {
      return value;
    }

    @Override
    public String toString() {
      return String.valueOf(value);
    }

    public static TransportTypeEnum fromValue(Integer value) {
      for (TransportTypeEnum b : TransportTypeEnum.values()) {
        if (b.value.equals(value)) {
          return b;
        }
      }
      throw new IllegalArgumentException("Unexpected value '" + value + "'");
    }

    public static class Adapter extends TypeAdapter<TransportTypeEnum> {
      @Override
      public void write(final JsonWriter jsonWriter, final TransportTypeEnum enumeration) throws IOException {
        jsonWriter.value(enumeration.getValue());
      }

      @Override
      public TransportTypeEnum read(final JsonReader jsonReader) throws IOException {
        Integer value =  jsonReader.nextInt();
        return TransportTypeEnum.fromValue(value);
      }
    }

    public static void validateJsonElement(JsonElement jsonElement) throws IOException {
      Integer value = jsonElement.getAsInt();
      TransportTypeEnum.fromValue(value);
    }
  }

  public static final String SERIALIZED_NAME_TRANSPORT_TYPE = "transport_type";
  @SerializedName(SERIALIZED_NAME_TRANSPORT_TYPE)
  private TransportTypeEnum transportType;

  public static final String SERIALIZED_NAME_APPLICABLE_LOCAL_DATE = "applicable_local_date";
  @SerializedName(SERIALIZED_NAME_APPLICABLE_LOCAL_DATE)
  private String applicableLocalDate;

  public static final String SERIALIZED_NAME_EXISTS = "exists";
  @SerializedName(SERIALIZED_NAME_EXISTS)
  private Boolean exists;

  public V3DynamoDbTimetable() {
  }

  public V3DynamoDbTimetable(
     String applicableLocalDate
  ) {
    this();
    this.applicableLocalDate = applicableLocalDate;
  }

  public V3DynamoDbTimetable tableName(String tableName) {
    this.tableName = tableName;
    return this;
  }

   /**
   * Name of corresponding table in DynamoDB.
   * @return tableName
  **/
  @javax.annotation.Nullable
  public String getTableName() {
    return tableName;
  }

  public void setTableName(String tableName) {
    this.tableName = tableName;
  }


  public V3DynamoDbTimetable parserVersion(Long parserVersion) {
    this.parserVersion = parserVersion;
    return this;
  }

   /**
   * Parser verison
   * @return parserVersion
  **/
  @javax.annotation.Nullable
  public Long getParserVersion() {
    return parserVersion;
  }

  public void setParserVersion(Long parserVersion) {
    this.parserVersion = parserVersion;
  }


  public V3DynamoDbTimetable parserMappingVersion(String parserMappingVersion) {
    this.parserMappingVersion = parserMappingVersion;
    return this;
  }

   /**
   * Diva Mapping Version used to load Parser into DynamoDB
   * @return parserMappingVersion
  **/
  @javax.annotation.Nullable
  public String getParserMappingVersion() {
    return parserMappingVersion;
  }

  public void setParserMappingVersion(String parserMappingVersion) {
    this.parserMappingVersion = parserMappingVersion;
  }


  public V3DynamoDbTimetable ptVersion(Long ptVersion) {
    this.ptVersion = ptVersion;
    return this;
  }

   /**
   * PT version
   * @return ptVersion
  **/
  @javax.annotation.Nullable
  public Long getPtVersion() {
    return ptVersion;
  }

  public void setPtVersion(Long ptVersion) {
    this.ptVersion = ptVersion;
  }


  public V3DynamoDbTimetable ptMappingVersion(String ptMappingVersion) {
    this.ptMappingVersion = ptMappingVersion;
    return this;
  }

   /**
   * Diva Mapping Version used to load PT into DynamoDB
   * @return ptMappingVersion
  **/
  @javax.annotation.Nullable
  public String getPtMappingVersion() {
    return ptMappingVersion;
  }

  public void setPtMappingVersion(String ptMappingVersion) {
    this.ptMappingVersion = ptMappingVersion;
  }


  public V3DynamoDbTimetable transportType(TransportTypeEnum transportType) {
    this.transportType = transportType;
    return this;
  }

   /**
   * A.k.a. Transport Mode (e.g. Train, Tram, Bus, V/Line, Nightrider)
   * @return transportType
  **/
  @javax.annotation.Nullable
  public TransportTypeEnum getTransportType() {
    return transportType;
  }

  public void setTransportType(TransportTypeEnum transportType) {
    this.transportType = transportType;
  }


   /**
   * Formated date string of applicable date
   * @return applicableLocalDate
  **/
  @javax.annotation.Nullable
  public String getApplicableLocalDate() {
    return applicableLocalDate;
  }



  public V3DynamoDbTimetable exists(Boolean exists) {
    this.exists = exists;
    return this;
  }

   /**
   * True if the named table has been created in DynamoDB (i.e. at least one departure record has been loaded),  or false if there are no records for this date and transport type.
   * @return exists
  **/
  @javax.annotation.Nullable
  public Boolean getExists() {
    return exists;
  }

  public void setExists(Boolean exists) {
    this.exists = exists;
  }



  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    V3DynamoDbTimetable v3DynamoDbTimetable = (V3DynamoDbTimetable) o;
    return Objects.equals(this.tableName, v3DynamoDbTimetable.tableName) &&
        Objects.equals(this.parserVersion, v3DynamoDbTimetable.parserVersion) &&
        Objects.equals(this.parserMappingVersion, v3DynamoDbTimetable.parserMappingVersion) &&
        Objects.equals(this.ptVersion, v3DynamoDbTimetable.ptVersion) &&
        Objects.equals(this.ptMappingVersion, v3DynamoDbTimetable.ptMappingVersion) &&
        Objects.equals(this.transportType, v3DynamoDbTimetable.transportType) &&
        Objects.equals(this.applicableLocalDate, v3DynamoDbTimetable.applicableLocalDate) &&
        Objects.equals(this.exists, v3DynamoDbTimetable.exists);
  }

  @Override
  public int hashCode() {
    return Objects.hash(tableName, parserVersion, parserMappingVersion, ptVersion, ptMappingVersion, transportType, applicableLocalDate, exists);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class V3DynamoDbTimetable {\n");
    sb.append("    tableName: ").append(toIndentedString(tableName)).append("\n");
    sb.append("    parserVersion: ").append(toIndentedString(parserVersion)).append("\n");
    sb.append("    parserMappingVersion: ").append(toIndentedString(parserMappingVersion)).append("\n");
    sb.append("    ptVersion: ").append(toIndentedString(ptVersion)).append("\n");
    sb.append("    ptMappingVersion: ").append(toIndentedString(ptMappingVersion)).append("\n");
    sb.append("    transportType: ").append(toIndentedString(transportType)).append("\n");
    sb.append("    applicableLocalDate: ").append(toIndentedString(applicableLocalDate)).append("\n");
    sb.append("    exists: ").append(toIndentedString(exists)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }


  public static HashSet<String> openapiFields;
  public static HashSet<String> openapiRequiredFields;

  static {
    // a set of all properties/fields (JSON key names)
    openapiFields = new HashSet<String>();
    openapiFields.add("table_name");
    openapiFields.add("parser_version");
    openapiFields.add("parser_mapping_version");
    openapiFields.add("pt_version");
    openapiFields.add("pt_mapping_version");
    openapiFields.add("transport_type");
    openapiFields.add("applicable_local_date");
    openapiFields.add("exists");

    // a set of required properties/fields (JSON key names)
    openapiRequiredFields = new HashSet<String>();
  }

 /**
  * Validates the JSON Element and throws an exception if issues found
  *
  * @param jsonElement JSON Element
  * @throws IOException if the JSON Element is invalid with respect to V3DynamoDbTimetable
  */
  public static void validateJsonElement(JsonElement jsonElement) throws IOException {
      if (jsonElement == null) {
        if (!V3DynamoDbTimetable.openapiRequiredFields.isEmpty()) { // has required fields but JSON element is null
          throw new IllegalArgumentException(String.format("The required field(s) %s in V3DynamoDbTimetable is not found in the empty JSON string", V3DynamoDbTimetable.openapiRequiredFields.toString()));
        }
      }

      Set<Map.Entry<String, JsonElement>> entries = jsonElement.getAsJsonObject().entrySet();
      // check to see if the JSON string contains additional fields
      for (Map.Entry<String, JsonElement> entry : entries) {
        if (!V3DynamoDbTimetable.openapiFields.contains(entry.getKey())) {
          throw new IllegalArgumentException(String.format("The field `%s` in the JSON string is not defined in the `V3DynamoDbTimetable` properties. JSON: %s", entry.getKey(), jsonElement.toString()));
        }
      }
        JsonObject jsonObj = jsonElement.getAsJsonObject();
      if ((jsonObj.get("table_name") != null && !jsonObj.get("table_name").isJsonNull()) && !jsonObj.get("table_name").isJsonPrimitive()) {
        throw new IllegalArgumentException(String.format("Expected the field `table_name` to be a primitive type in the JSON string but got `%s`", jsonObj.get("table_name").toString()));
      }
      if ((jsonObj.get("parser_mapping_version") != null && !jsonObj.get("parser_mapping_version").isJsonNull()) && !jsonObj.get("parser_mapping_version").isJsonPrimitive()) {
        throw new IllegalArgumentException(String.format("Expected the field `parser_mapping_version` to be a primitive type in the JSON string but got `%s`", jsonObj.get("parser_mapping_version").toString()));
      }
      if ((jsonObj.get("pt_mapping_version") != null && !jsonObj.get("pt_mapping_version").isJsonNull()) && !jsonObj.get("pt_mapping_version").isJsonPrimitive()) {
        throw new IllegalArgumentException(String.format("Expected the field `pt_mapping_version` to be a primitive type in the JSON string but got `%s`", jsonObj.get("pt_mapping_version").toString()));
      }
      // validate the optional field `transport_type`
      if (jsonObj.get("transport_type") != null && !jsonObj.get("transport_type").isJsonNull()) {
        TransportTypeEnum.validateJsonElement(jsonObj.get("transport_type"));
      }
      if ((jsonObj.get("applicable_local_date") != null && !jsonObj.get("applicable_local_date").isJsonNull()) && !jsonObj.get("applicable_local_date").isJsonPrimitive()) {
        throw new IllegalArgumentException(String.format("Expected the field `applicable_local_date` to be a primitive type in the JSON string but got `%s`", jsonObj.get("applicable_local_date").toString()));
      }
  }

  public static class CustomTypeAdapterFactory implements TypeAdapterFactory {
    @SuppressWarnings("unchecked")
    @Override
    public <T> TypeAdapter<T> create(Gson gson, TypeToken<T> type) {
       if (!V3DynamoDbTimetable.class.isAssignableFrom(type.getRawType())) {
         return null; // this class only serializes 'V3DynamoDbTimetable' and its subtypes
       }
       final TypeAdapter<JsonElement> elementAdapter = gson.getAdapter(JsonElement.class);
       final TypeAdapter<V3DynamoDbTimetable> thisAdapter
                        = gson.getDelegateAdapter(this, TypeToken.get(V3DynamoDbTimetable.class));

       return (TypeAdapter<T>) new TypeAdapter<V3DynamoDbTimetable>() {
           @Override
           public void write(JsonWriter out, V3DynamoDbTimetable value) throws IOException {
             JsonObject obj = thisAdapter.toJsonTree(value).getAsJsonObject();
             elementAdapter.write(out, obj);
           }

           @Override
           public V3DynamoDbTimetable read(JsonReader in) throws IOException {
             JsonElement jsonElement = elementAdapter.read(in);
             validateJsonElement(jsonElement);
             return thisAdapter.fromJsonTree(jsonElement);
           }

       }.nullSafe();
    }
  }

 /**
  * Create an instance of V3DynamoDbTimetable given an JSON string
  *
  * @param jsonString JSON string
  * @return An instance of V3DynamoDbTimetable
  * @throws IOException if the JSON string is invalid with respect to V3DynamoDbTimetable
  */
  public static V3DynamoDbTimetable fromJson(String jsonString) throws IOException {
    return JSON.getGson().fromJson(jsonString, V3DynamoDbTimetable.class);
  }

 /**
  * Convert an instance of V3DynamoDbTimetable to an JSON string
  *
  * @return JSON string
  */
  public String toJson() {
    return JSON.getGson().toJson(this);
  }
}

File: swagger-3/python/swagger_client/models/v3_dynamo_db_timetable.py
# coding: utf-8

"""
    PTV Timetable API - Version 3

    The PTV Timetable API provides direct access to Public Transport Victoria's public transport timetable data.    The API returns scheduled timetable, route and stop data for all metropolitan and regional train, tram and bus services in Victoria, including Night Network(Night Train and Night Tram data are included in metropolitan train and tram services data, respectively, whereas Night Bus is a separate route type).    The API also returns real-time data for metropolitan train, tram and bus services (where this data is made available to PTV), as well as disruption information, stop facility information, and access to myki ticket outlet data.    This Swagger is for Version 3 of the PTV Timetable API. By using this documentation you agree to comply with the licence and terms of service.    Train timetable data is updated daily, while the remaining data is updated weekly, taking into account any planned timetable changes (for example, due to holidays or planned disruptions). The PTV timetable API is the same API used by PTV for its apps. To access the most up to date data PTV has (including real-time data) you must use the API dynamically.    You can access the PTV Timetable API through a HTTP or HTTPS interface, as follows:        base URL / version number / API name / query string  The base URL is either:    *  http://timetableapi.ptv.vic.gov.au  or    *  https://timetableapi.ptv.vic.gov.au    The Swagger JSON file is available at http://timetableapi.ptv.vic.gov.au/swagger/docs/v3    Frequently asked questions are available on the PTV website at http://ptv.vic.gov.au/apifaq    Links to the following information are also provided on the PTV website at http://ptv.vic.gov.au/ptv-timetable-api/  * How to register for an API key and calculate a signature  * PTV Timetable API V2 to V3 Migration Guide  * PTV Timetable API Data Quality Statement    All information about how to use the API is in this documentation. PTV cannot provide technical support for the API.    # noqa: E501

    OpenAPI spec version: v3
    
    Generated by: https://github.com/swagger-api/swagger-codegen.git
"""

import pprint
import re  # noqa: F401

import six

class V3DynamoDbTimetable(object):
    """NOTE: This class is auto generated by the swagger code generator program.

    Do not edit the class manually.
    """
    """
    Attributes:
      swagger_types (dict): The key is attribute name
                            and the value is attribute type.
      attribute_map (dict): The key is attribute name
                            and the value is json key in definition.
    """
    swagger_types = {
        'table_name': 'str',
        'parser_version': 'int',
        'parser_mapping_version': 'str',
        'pt_version': 'int',
        'pt_mapping_version': 'str',
        'transport_type': 'int',
        'applicable_local_date': 'str',
        'exists': 'bool'
    }

    attribute_map = {
        'table_name': 'table_name',
        'parser_version': 'parser_version',
        'parser_mapping_version': 'parser_mapping_version',
        'pt_version': 'pt_version',
        'pt_mapping_version': 'pt_mapping_version',
        'transport_type': 'transport_type',
        'applicable_local_date': 'applicable_local_date',
        'exists': 'exists'
    }

    def __init__(self, table_name=None, parser_version=None, parser_mapping_version=None, pt_version=None, pt_mapping_version=None, transport_type=None, applicable_local_date=None, exists=None):  # noqa: E501
        """V3DynamoDbTimetable - a model defined in Swagger"""  # noqa: E501
        self._table_name = None
        self._parser_version = None
        self._parser_mapping_version = None
        self._pt_version = None
        self._pt_mapping_version = None
        self._transport_type = None
        self._applicable_local_date = None
        self._exists = None
        self.discriminator = None
        if table_name is not None:
            self.table_name = table_name
        if parser_version is not None:
            self.parser_version = parser_version
        if parser_mapping_version is not None:
            self.parser_mapping_version = parser_mapping_version
        if pt_version is not None:
            self.pt_version = pt_version
        if pt_mapping_version is not None:
            self.pt_mapping_version = pt_mapping_version
        if transport_type is not None:
            self.transport_type = transport_type
        if applicable_local_date is not None:
            self.applicable_local_date = applicable_local_date
        if exists is not None:
            self.exists = exists

    @property
    def table_name(self):
        """Gets the table_name of this V3DynamoDbTimetable.  # noqa: E501

        Name of corresponding table in DynamoDB.  # noqa: E501

        :return: The table_name of this V3DynamoDbTimetable.  # noqa: E501
        :rtype: str
        """
        return self._table_name

    @table_name.setter
    def table_name(self, table_name):
        """Sets the table_name of this V3DynamoDbTimetable.

        Name of corresponding table in DynamoDB.  # noqa: E501

        :param table_name: The table_name of this V3DynamoDbTimetable.  # noqa: E501
        :type: str
        """

        self._table_name = table_name

    @property
    def parser_version(self):
        """Gets the parser_version of this V3DynamoDbTimetable.  # noqa: E501

        Parser verison  # noqa: E501

        :return: The parser_version of this V3DynamoDbTimetable.  # noqa: E501
        :rtype: int
        """
        return self._parser_version

    @parser_version.setter
    def parser_version(self, parser_version):
        """Sets the parser_version of this V3DynamoDbTimetable.

        Parser verison  # noqa: E501

        :param parser_version: The parser_version of this V3DynamoDbTimetable.  # noqa: E501
        :type: int
        """

        self._parser_version = parser_version

    @property
    def parser_mapping_version(self):
        """Gets the parser_mapping_version of this V3DynamoDbTimetable.  # noqa: E501

        Diva Mapping Version used to load Parser into DynamoDB  # noqa: E501

        :return: The parser_mapping_version of this V3DynamoDbTimetable.  # noqa: E501
        :rtype: str
        """
        return self._parser_mapping_version

    @parser_mapping_version.setter
    def parser_mapping_version(self, parser_mapping_version):
        """Sets the parser_mapping_version of this V3DynamoDbTimetable.

        Diva Mapping Version used to load Parser into DynamoDB  # noqa: E501

        :param parser_mapping_version: The parser_mapping_version of this V3DynamoDbTimetable.  # noqa: E501
        :type: str
        """

        self._parser_mapping_version = parser_mapping_version

    @property
    def pt_version(self):
        """Gets the pt_version of this V3DynamoDbTimetable.  # noqa: E501

        PT version  # noqa: E501

        :return: The pt_version of this V3DynamoDbTimetable.  # noqa: E501
        :rtype: int
        """
        return self._pt_version

    @pt_version.setter
    def pt_version(self, pt_version):
        """Sets the pt_version of this V3DynamoDbTimetable.

        PT version  # noqa: E501

        :param pt_version: The pt_version of this V3DynamoDbTimetable.  # noqa: E501
        :type: int
        """

        self._pt_version = pt_version

    @property
    def pt_mapping_version(self):
        """Gets the pt_mapping_version of this V3DynamoDbTimetable.  # noqa: E501

        Diva Mapping Version used to load PT into DynamoDB  # noqa: E501

        :return: The pt_mapping_version of this V3DynamoDbTimetable.  # noqa: E501
        :rtype: str
        """
        return self._pt_mapping_version

    @pt_mapping_version.setter
    def pt_mapping_version(self, pt_mapping_version):
        """Sets the pt_mapping_version of this V3DynamoDbTimetable.

        Diva Mapping Version used to load PT into DynamoDB  # noqa: E501

        :param pt_mapping_version: The pt_mapping_version of this V3DynamoDbTimetable.  # noqa: E501
        :type: str
        """

        self._pt_mapping_version = pt_mapping_version

    @property
    def transport_type(self):
        """Gets the transport_type of this V3DynamoDbTimetable.  # noqa: E501

        A.k.a. Transport Mode (e.g. Train, Tram, Bus, V/Line, Nightrider)  # noqa: E501

        :return: The transport_type of this V3DynamoDbTimetable.  # noqa: E501
        :rtype: int
        """
        return self._transport_type

    @transport_type.setter
    def transport_type(self, transport_type):
        """Sets the transport_type of this V3DynamoDbTimetable.

        A.k.a. Transport Mode (e.g. Train, Tram, Bus, V/Line, Nightrider)  # noqa: E501

        :param transport_type: The transport_type of this V3DynamoDbTimetable.  # noqa: E501
        :type: int
        """
        allowed_values = [0, 1, 2, 3, 4]  # noqa: E501
        if transport_type not in allowed_values:
            raise ValueError(
                "Invalid value for `transport_type` ({0}), must be one of {1}"  # noqa: E501
                .format(transport_type, allowed_values)
            )

        self._transport_type = transport_type

    @property
    def applicable_local_date(self):
        """Gets the applicable_local_date of this V3DynamoDbTimetable.  # noqa: E501

        Formated date string of applicable date  # noqa: E501

        :return: The applicable_local_date of this V3DynamoDbTimetable.  # noqa: E501
        :rtype: str
        """
        return self._applicable_local_date

    @applicable_local_date.setter
    def applicable_local_date(self, applicable_local_date):
        """Sets the applicable_local_date of this V3DynamoDbTimetable.

        Formated date string of applicable date  # noqa: E501

        :param applicable_local_date: The applicable_local_date of this V3DynamoDbTimetable.  # noqa: E501
        :type: str
        """

        self._applicable_local_date = applicable_local_date

    @property
    def exists(self):
        """Gets the exists of this V3DynamoDbTimetable.  # noqa: E501

        True if the named table has been created in DynamoDB (i.e. at least one departure record has been loaded),  or false if there are no records for this date and transport type.  # noqa: E501

        :return: The exists of this V3DynamoDbTimetable.  # noqa: E501
        :rtype: bool
        """
        return self._exists

    @exists.setter
    def exists(self, exists):
        """Sets the exists of this V3DynamoDbTimetable.

        True if the named table has been created in DynamoDB (i.e. at least one departure record has been loaded),  or false if there are no records for this date and transport type.  # noqa: E501

        :param exists: The exists of this V3DynamoDbTimetable.  # noqa: E501
        :type: bool
        """

        self._exists = exists

    def to_dict(self):
        """Returns the model properties as a dict"""
        result = {}

        for attr, _ in six.iteritems(self.swagger_types):
            value = getattr(self, attr)
            if isinstance(value, list):
                result[attr] = list(map(
                    lambda x: x.to_dict() if hasattr(x, "to_dict") else x,
                    value
                ))
            elif hasattr(value, "to_dict"):
                result[attr] = value.to_dict()
            elif isinstance(value, dict):
                result[attr] = dict(map(
                    lambda item: (item[0], item[1].to_dict())
                    if hasattr(item[1], "to_dict") else item,
                    value.items()
                ))
            else:
                result[attr] = value
        if issubclass(V3DynamoDbTimetable, dict):
            for key, value in self.items():
                result[key] = value

        return result

    def to_str(self):
        """Returns the string representation of the model"""
        return pprint.pformat(self.to_dict())

    def __repr__(self):
        """For `print` and `pprint`"""
        return self.to_str()

    def __eq__(self, other):
        """Returns true if both objects are equal"""
        if not isinstance(other, V3DynamoDbTimetable):
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        """Returns true if both objects are not equal"""
        return not self == other
